### PR TITLE
Add a build pipeline for prow

### PR DESCRIPTION
**Pull request by 0xmichalis**
@openshift/ci-administrators

### CODE REVIEW

**Comment by kargakis** [comment]
@openshift/ci-administrators seems there is an issue with pulls from the ci registry 

<details>
$ docker pull docker-registry-default.svc.ci.openshift.org/ci/busybox:latest
Pulling repository docker-registry-default.svc.ci.openshift.org/ci/busybox
Error: Status 503 trying to pull repository ci/busybox: "<html>\n  <head>\n    <meta name=\"viewport\" content=\"width=device-width, initial-scale=1\">\n\n  <style type=\"text/css\">\n  /*!\n   * Bootstrap v3.3.5 (http://getbootstrap.com)\n   * Copyright 2011-2015 Twitter, Inc.\n   * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)\n   */\n  /*! normalize.css v3.0.3 | MIT License | github.com/necolas/normalize.css */\n  html {\n    font-family: sans-serif;\n    -ms-text-size-adjust: 100%;\n    -webkit-text-size-adjust: 100%;\n  }\n  body {\n    margin: 0;\n  }\n  h1 {\n    font-size: 1.7em;\n    font-weight: 400;\n    line-height: 1.3;\n    margin: 0.68em 0;\n  }\n  * {\n    -webkit-box-sizing: border-box;\n    -moz-box-sizing: border-box;\n    box-sizing: border-box;\n  }\n  *:before,\n  *:after {\n    -webkit-box-sizing: border-box;\n    -moz-box-sizing: border-box;\n    box-sizing: border-box;\n  }\n  html {\n    -webkit-tap-highlight-color: rgba(0, 0, 0, 0);\n  }\n  body {\n    font-family: \"Helvetica Neue\", Helvetica, Arial, sans-serif;\n    line-height: 1.66666667;\n    font-size: 13px;\n    color: #333333;\n    background-color: #ffffff;\n    margin: 2em 1em;\n  }\n  p {\n    margin: 0 0 10px;\n    font-size: 13px;\n  }\n  .alert.alert-info {\n    padding: 15px;\n    margin-bottom: 20px;\n    border: 1px solid transparent;\n    background-color: #f5f5f5;\n    border-color: #8b8d8f;\n    color: #363636;\n    margin-top: 30px;\n  }\n  .alert p {\n    padding-left: 35px;\n  }\n  a {\n    color: #0088ce;\n  }\n\n  ul {\n    position: relative;\n    padding-left: 51px;\n  }\n  p.info {\n    position: relative;\n    font-size: 15px;\n    margin-bottom: 10px;\n  }\n  p.info:before, p.info:after {\n    content: \"\";\n    position: absolute;\n    top: 9%;\n    left: 0;\n  }\n  p.info:before {\n    content: \"i\";\n    left: 3px;\n    width: 20px;\n    height: 20px;\n    font-family: serif;\n    font-size: 15px;\n    font-weight: bold;\n    line-height: 21px;\n    text-align: center;\n    color: #fff;\n    background: #4d5258;\n    border-radius: 16px;\n  }\n\n  @media (min-width: 768px) {\n    body {\n      margin: 4em 3em;\n    }\n    h1 {\n      font-size: 2.15em;}\n  }\n\n  </style>\n  </head>\n  <body>\n    <div>\n      <h1>Application is not available</h1>\n      <p>The application is currently not serving requests at this endpoint. It may not have been started or is still starting.</p>\n\n      <div class=\"alert alert-info\">\n        <p class=\"info\">\n          Possible reasons you are seeing this page:\n        </p>\n        <ul>\n          <li>\n            <strong>The host doesn't exist.</strong>\n            Make sure the hostname was typed correctly and that a route matching this hostname exists.\n          </li>\n          <li>\n            <strong>The host exists, but doesn't have a matching path.</strong>\n            Check if the URL path was typed correctly and that the route was created using the desired path.\n          </li>\n          <li>\n            <strong>Route and path matches, but all pods are down.</strong>\n            Make sure that the resources exposed by this route (pods, services, deployment configs, etc) have at least one pod running.\n          </li>\n        </ul>\n      </div>\n    </div>\n  </body>\n</html>\n"
</details>

File: cluster/ci/config/prow/prow_images.yaml [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+      dockerfile: |
+        FROM openshift/origin-release:golang-1.8
+
+        COPY . /go/src/k8s.io/test-infra
+        RUN cd /go/src/k8s.io/test-infra && \
+            godep restore && \
+            cd prow && \
+            mkdir -p /static && \
+            cp -r cmd/deck/static/* /static && \
+            CGO_ENABLED=0 go install -a -installsuffix cgo -ldflags '-w' ./cmd/... && \
```
Comment (kargakis) [inline cluster/ci/config/prow/prow_images.yaml]: Eventually, each component should have its own build chain. I am thinking of fixing it in this PR..

**Comment by stevekuznetsov** [comment]
Is the registry deployment healthy in `namespaces/default`?

**Comment by kargakis** [comment]
It's fine
```
x - - [15/Jul/2017:19:45:53 +0000] "GET /healthz HTTP/2.0" 200 0 "" "Go-http-client/2.0"
x- - [15/Jul/2017:19:46:03 +0000] "GET /healthz HTTP/2.0" 200 0 "" "Go-http-client/2.0"
x - - [15/Jul/2017:19:46:03 +0000] "GET /healthz HTTP/2.0" 200 0 "" "Go-http-client/2.0"
x- - [15/Jul/2017:19:46:13 +0000] "GET /healthz HTTP/2.0" 200 0 "" "Go-http-client/2.0"
x - - [15/Jul/2017:19:46:13 +0000] "GET /healthz HTTP/2.0" 200 0 "" "Go-http-client/2.0"
x - - [15/Jul/2017:19:46:23 +0000] "GET /healthz HTTP/2.0" 200 0 "" "Go-http-client/2.0"
x - - [15/Jul/2017:19:46:23 +0000] "GET /healthz HTTP/2.0" 200 0 "" "Go-http-client/2.0"
```

**Comment by kargakis** [comment]
There are some authorization-related errors in the log but not clear if they are related. I can't see anything when doing the docker pull.

**Comment by stevekuznetsov** [comment]
Where are you pulling from? Could it be a routing issue?

**Comment by kargakis** [comment]
Getting this error both from my host and from GKE.

**Pull request merged**